subsys/net/lib/conn_mgr/conn_mgr.c

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(conn_mgr, CONFIG_NET_CONNECTION_MANAGER_LOG_LEVEL);
+
+#include <init.h>
+#include <kernel.h>
+#include <errno.h>
+#include <net/net_core.h>
+#include <net/net_if.h>
+#include <net/net_mgmt.h>
+
+#include <conn_mgr.h>
+
+u16_t iface_states[CONN_MGR_IFACE_MAX];
```

Comment:
indeed, much better

Reply:
Only caveat is that if there are two network interfaces where the other would only have IPv6 support and the other IPv4 support, then the MAX value of this would be 1 instead of intended 2. Fortunately this scenario is not very realistic. There is no really good solution for figuring out how many network interfaces we have at compile time.

Reply:
yes and since the type of this array is u16_t it's not going to eat a lot of memory